Assets/Scripts/DayNightCycle/UpdateClock.cs

```diff
@@ -0,0 +1,11 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class UpdateClock : MonoBehaviour
+{
+    void Update()
+    {
+        TimeStamps.ClockUpdate(Time.deltaTime);
```

Comment:
my only worry with this is having multiple update clocks on accident in scene. could use a static constructor to handle scene initialization?

so there's a technique i just found / figured out to do singletons statically.

i added the below code to the timestamps class, set ClockUpdate to private, and deleted UpdateClock.cs
``` c#
private class UpdateClock : MonoBehaviour
    {
        public delegate void Tick(float deltaTime);
        public Tick tick;
        void Update()
        {
            tick(Time.deltaTime);
        }
    }

    private static UpdateClock Ticker;
    public static bool ClockIsTicking { 
        get { return timeRunning; } 
        set 
        { 
            timeRunning = value;
            Ticker.enabled = value;
        } 
    }

    static TimeStamps()
    {
        if (!Ticker)
        {
            var go = new GameObject();
            go.AddComponent<UpdateClock>();
            UnityEngine.Object.DontDestroyOnLoad(go);
            Ticker = go.GetComponent<UpdateClock>();
            Ticker.tick = ClockUpdate;
        }
        timeRunning = true;
        Ticker.enabled = true;
    }
```

im not pushing anything, just thought it was neat and i'll be using this tech in my other manager class

Reply:
I see, will give it some thought and implement it, wanted to avoid singletons, but my code is already a singleton

Reply:
The idea was to add that line of code to the (i think) already existing Game Manager. Since I supposed it would be on a prefab, or someone would already be working on it, I didn't add it. 

My bad for not saying it.